clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -536,12 +543,23 @@ public void onResponse(final ClientResponse response) {
                         continue;
                     }
 
-                    if (error == Errors.COORDINATOR_NOT_AVAILABLE ||
+                    if (error == Errors.GROUP_AUTHORIZATION_FAILED) {
+                        future.completeExceptionally(GroupAuthorizationException.forGroupId(groupId));
+                        return;
+                    } else if (error == Errors.COORDINATOR_NOT_AVAILABLE ||
                         error == Errors.NOT_COORDINATOR ||
                         error == Errors.REQUEST_TIMED_OUT) {
                         coordinatorRequestManager.markCoordinatorUnknown(error.message(), currentTimeMs);
                         maybeRetry(currentTimeMs, error.exception());
                         return;
+                    } else if (error == Errors.FENCED_INSTANCE_ID) {
+                        log.info("OffsetCommit failed due to group instance id {} fenced: {}", groupInstanceId, error.message());
+                        future.completeExceptionally(new CommitFailedException());
```

Comment:
nit: While we are here, should we put an error msg?

Reply:
This one has not been addressed yet.

Reply:
uhm I had addressed it by adding `log.error` instead of `log.info`, but maybe I misunderstood your comment. Were you suggesting something different?

Reply:
Ah. I was referring to adding a message to `new CommitFailedException()`. Sorry for the confusion.

Reply:
Oh maybe just a confusion...I notice now that on the main PR page it still shows the `log.info`, but we do have the `log.error` I introduced in [this](https://github.com/apache/kafka/pull/15202/commits/f7b76fd75de6615ac089c6c8ff24166fc5683a2d) previous commit , so maybe just github tricking us?

Reply:
got it. Message added to the exception too.